psd-web/app/models/organisation.rb

```diff
@@ -1,32 +1,13 @@
-class Organisation < ActiveHash::Base
-  include ActiveHash::Associations
-  include ActiveHashSafeLoadable
-
-  field :id
-  field :name
-  field :path
-
+class Organisation < ApplicationRecord
   has_many :users, dependent: :nullify
   has_many :teams, dependent: :nullify
 
-  def self.load(force: false)
-    begin
-      self.safe_load(KeycloakClient.instance.all_organisations(force: force), data_name: "organisations")
-    rescue StandardError => e
-      Rails.logger.error "Failed to fetch organisations from Keycloak: #{e.message}"
-      self.data = nil
-    end
-  end
-
-  def self.all(options = {})
-    self.load
+  validates :id, presence: true, uuid: true
 
-    if options.has_key?(:conditions)
-      where(options[:conditions])
-    else
-      @records ||= []
+  def self.load_from_keycloak(orgs = KeycloakClient.instance.all_organisations)
+    orgs.each do |org|
+      record = find_or_create_by(id: org[:id])
```

Comment:
```suggestion
      record = create_or_find_by!(id: org[:id])
```
no biggy but find_or_create_by is not atomic, create_or_find_by is

Reply:
Thanks, good tip.

Reply:
`create_or_find_by!` was added in Rails 6 but we are running 5.2.3 still. However I will add validations and use `find_or_create_by!` as it will prevent defective data being persisted in our app.